src/Parallel/dominatingset/minimal_dom_set.jl

```diff
@@ -1,14 +1,19 @@
 """
     dominating_set(g, reps, MinimalDominatingSet(); parallel=:threads, seed=-1)
 
-Perform [`LightGraphs.dominating_set(g, MinimalDominatingSet())`](@ref) `reps` times in parallel 
+Perform [`LightGraphs.dominating_set(g, MinimalDominatingSet())`](@ref) `reps` times in parallel
 and return the solution with the fewest vertices.
 
 ### Optional Arguements
 - `parallel=:threads`: If `parallel=:distributed` then the multiprocessor implementation is
 used. This implementation is more efficient if `reps` is large.
 - If `seed >= 0`, a random generator of each process/thread is seeded with this value.
 """
-dominating_set(g::AbstractGraph{T}, reps::Integer, alg::MinimalDominatingSet; parallel=:threads, seed=-1) where T <: Integer = 
-LightGraphs.Parallel.generate_reduce(g, (g::AbstractGraph{T})->LightGraphs.dominating_set(g, alg; seed=seed), 
-(x::Vector{T}, y::Vector{T})->length(x)<length(y), reps; parallel=parallel)
+function dominating_set(g::AbstractGraph{T}, reps::Integer, alg::MinimalDominatingSet; parallel=:threads, seed=-1) where T <: Integer
```

Comment:
Should this take an RNG instead of a seed?

Reply:
good catch, however, we can't provide a rng directly for race conditions. So I suggest providing a function that gives a rng, like `() -> MersenneTwister(1)`